ui/js/signup.js

```diff
@@ -1,29 +1,38 @@
-// const formData = document.getElementById('formData').addEventListener('submit', signUpUser);
-const signUP = document.getElementById('signUP')
-
-function signUpUser(event){
+window.onload = function () {
+    const formData = document.getElementById('formData')
+    formData.addEventListener('submit', signUpUser);
+}
+function signUpUser(event) {
     /*
     Function to perform user signup
     */
-   event.preventDefault();
+    event.preventDefault();
+
+    let firstname = document.getElementById('firstname').value
+    let lastname = document.getElementById('lastname').value
+    let othername = document.getElementById('othername').value
+    let email = document.getElementById('email').value
+    let phoneNumber = document.getElementById('phone').value
+    let username = document.getElementById('username').value
+    let password = document.getElementById('password').value
 
-   fetch('https://questioner-system.herokuapp.com/api/v2/signup', {
-       method: 'POST',
-       body: JSON.stringify({
-            "firstname": document.getElementById('firstname').value,
-            "lastname": document.getElementById('lastname').value,
-            "othername": document.getElementById('othername').value,
-            "email": document.getElementById('email').value,
-            "phone": document.getElementById('phone').value,
-            "username": document.getElementById('username').value,
-            "password": document.getElementById('password').value
-       }),
-       headers: {
-        'Content-Type': 'application/json'
-      }
-   })
-   .then(res => res.json())
-   .then((data => console.log('Success:', JSON.stringify(data))))
-   .catch(error => console.log('Error', error))
+    fetch('http://127.0.0.1:5000/api/v2/auth/signup', {
```

Comment:
@a-braham This is good work, though you could have like a base file, where you have the URL that you will be importing to other files to avoid repeating this `http://127.0.0.1:5000/api/v2/` throughout your code

Reply:
Awesome..that's nice idea..let me see how I can refactor my code

Reply:
Resolving in bug a branch